datalad/support/tests/test_sshconnector.py

```diff
@@ -74,3 +74,39 @@ def test_ssh_manager_close():
 
     ok_(not exists(opj(manager.socket_dir, 'localhost')))
     ok_(not exists(opj(manager.socket_dir, 'datalad-test')))
+
+
+@skip_ssh
+@with_tempfile(mkdir=True)
+@with_tempfile(content="one")
+@with_tempfile(content="two")
+def test_ssh_copy(sourcedir, sourcefile1, sourcefile2):
+
+    remote_url = 'ssh://localhost'
+    manager = SSHManager()
+    ssh = manager.get_connection(remote_url)
+    ssh.open()
```

Comment:
yeap. the idea is to be able to do

```python
with ssh:
    ssh("remote command")
    ssh.copy("file", "dest")
```

and then possibly later we would also look into improving the tandem of manager + connection -- imho we might better be off without explicit separation into two and doing everything within one class.  Now establishing a new ssh connection is a bit cumbersome (get a manager, ask for connection, connect) -- imho it could/should be easier and all the connection pulling could be done within the same class, using class variable (thus providing desired singleton implementation) to store pooled connections... so I could just do

```
from datalad.support.ssh import SSH
with SSH("myhost", port=12) as ssh:
     ssh.copy("here", "there")
```
or
```
with SSH("ssh://yarikoptic@thehost:2222") as ssh:
    ssh.copy("here", "there")
```
And which atexit would just call classmethod `SSH.close_all_connections()`

but let's make all that next separate PR.  I am merging this one meanwhile